extension/packages/nextjs/components/FlagTracker.tsx

```diff
@@ -0,0 +1,70 @@
+"use client";
+
+import { useAccount } from "wagmi";
+import { useScaffoldEventHistory } from "~~/hooks/scaffold-eth";
+
+export const FlagTracker = () => {
+  const { address: connectedAddress } = useAccount();
+  const { data: flagEvents } = useScaffoldEventHistory({
+    contractName: "NFTFlags",
+    eventName: "FlagMinted",
+    fromBlock: 0n,
+    watch: true,
+  });
+
+  // Filter flags for the connected user
+  const userMintedFlags =
```

Comment:
You can use the filters parameter on useScaffoldEventHistory to get only the flags from the connected address..